src/main/java/org/elasticsearch/common/io/Channels.java

```diff
@@ -87,6 +87,22 @@ public static int readFromFileChannel(FileChannel channel, long channelPosition,
         return readFromFileChannel(channel, channelPosition, buffer);
     }
 
+
+    /**
+     * read from a file channel into a byte buffer, starting at a certain position.  An EOFException will be thrown if you
+     * attempt to read beyond the end of file.
+     *
+     * @param channel         channel to read from
+     * @param channelPosition position to read from
+     * @param dest            destination {@link java.nio.ByteBuffer} to put data in
+     */
+    public static void readFromFileChannelWithEofException(FileChannel channel, long channelPosition, ByteBuffer dest) throws IOException {
```

Comment:
we already have `readFromFileChannelWithEofException` . agreed it's long but rather do it in another change..